docs/mypy.md

```diff
@@ -0,0 +1,30 @@
+# mypy static checker
+
+Zulip now uses mypy. mypy is a compile-time type checker for python.
+For information about what mypy is, visit [http://mypy-lang.org/](http://mypy-lang.org/).
+
+mypy's type checking has been added to Zulip's Travis CI checks.
+
+## Running mypy on Zulip's code
+
+To run mypy on Zulip's code, you'll need to make sure that mypy is installed.
+mypy only runs on python 3. Zulip doesn't use the latest version of mypy available on pypi.
```

Comment:
Let's make this more clear about what's going on: "Zulip doesn't use the latest version of mypy available on pypi." should say "Since mypy is a new tool under rapid development, Zulip is using a pinned version of mypy from upstream Git rather than tracking the (older) latest mypy release on pypi."